pkg/install/app/vaultconfigdeps.go

```diff
@@ -0,0 +1,239 @@
+	vaultTokenEnvVarName      = "RELAY_OPERATOR_VAULT_INIT_VAULT_TOKEN"
+	vaultUnsealKeyEnvVarName  = "RELAY_OPERATOR_VAULT_INIT_VAULT_UNSEAL_KEY"
+
+	logServicePathEnvVar            = "RELAY_OPERATOR_VAULT_INIT_LOG_SERVICE_PATH"
+	logServiceVaultAgentRoleEnvVar  = "RELAY_OPERATOR_VAULT_INIT_LOG_SERVICE_VAULT_AGENT_ROLE"
+	metadataAPIVaultAgentRoleEnvVar = "RELAY_OPERATOR_VAULT_INIT_METADATA_API_VAULT_AGENT_ROLE"
+	operatorVaultAgentRoleEnvVar    = "RELAY_OPERATOR_VAULT_INIT_OPERATOR_VAULT_AGENT_ROLE"
+	tenantPathEnvVar                = "RELAY_OPERATOR_VAULT_INIT_TENANT_PATH"
+	transitKeyEnvVar                = "RELAY_OPERATOR_VAULT_INIT_TRANSIT_KEY"
+	transitPathEnvVar               = "RELAY_OPERATOR_VAULT_INIT_TRANSIT_PATH"
+)
+
+type VaultConfigDeps struct {
+	Core                *obj.Core
+	ConfigJob           *batchv1obj.Job
+	JWTSigningKeySecret *corev1obj.Secret
+	OwnerConfigMap      *corev1obj.ConfigMap
+	Labels              map[string]string
+}
+
+func (vd *VaultConfigDeps) Load(ctx context.Context, cl client.Client) (bool, error) {
+	if _, err := vd.Core.Load(ctx, cl); err != nil {
+		return false, err
+	}
+
+	key := helper.SuffixObjectKey(vd.Core.Key, vaultInitIdentifier)
+
+	vd.OwnerConfigMap = corev1obj.NewConfigMap(helper.SuffixObjectKey(key, "owner"))
+
+	vd.ConfigJob = batchv1obj.NewJob(key)
+
+	loaders := lifecycle.Loaders{
+		vd.OwnerConfigMap,
+		vd.ConfigJob,
+		vd.JWTSigningKeySecret,
+	}
+
+	if _, err := loaders.Load(ctx, cl); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (vd *VaultConfigDeps) Owned(ctx context.Context, owner lifecycle.TypedObject) error {
+	return helper.Own(vd.OwnerConfigMap.Object, owner)
+}
+
+func (vd *VaultConfigDeps) Persist(ctx context.Context, cl client.Client) error {
+	if err := vd.OwnerConfigMap.Persist(ctx, cl); err != nil {
+		return err
+	}
+
+	objs := []lifecycle.OwnablePersister{
+		vd.ConfigJob,
+	}
+
+	for _, obj := range objs {
+		if err := vd.OwnerConfigMap.Own(ctx, obj); err != nil {
+			return err
+		}
+
+		if err := obj.Persist(ctx, cl); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (vd *VaultConfigDeps) Configure(ctx context.Context) error {
+	err := DependencyManager.SetDependencyOf(
+		vd.OwnerConfigMap.Object,
+		lifecycle.TypedObject{
+			Object: vd.Core.Object,
+			GVK:    v1alpha1.RelayCoreKind,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	lafs := []lifecycle.LabelAnnotatableFrom{
+		vd.ConfigJob,
+	}
+
+	for _, laf := range lafs {
+		for label, value := range vd.Labels {
+			lifecycle.Label(ctx, laf, label, value)
+		}
+	}
+
+	ConfigureVaultConfigJob(
+		vd.Core.Key,
+		vd.Core.Object.Spec.LogService,
+		vd.Core.Object.Spec.MetadataAPI,
+		vd.Core.Object.Spec.Operator,
+		vd.Core.Object.Spec.Vault,
+		vd.ConfigJob, vd.JWTSigningKeySecret)
+
+	return nil
+}
+
+func NewVaultConfigDeps(c *obj.Core, jwt *corev1obj.Secret) *VaultConfigDeps {
+	vd := &VaultConfigDeps{
+		Core:                c,
```

Comment:
For consistency's sake I might create the `OwnerConfigMap` and `ConfigJob` instances here instead of in `Load()`.